src/AzureRMTools.ts

```diff
@@ -92,6 +93,30 @@ export class AzureRMTools {
         registerCommand("azurerm-vscode-tools.reloadSchemas", async () => {
             await reloadSchemas();
         });
+        registerCommand("azurerm-vscode-tools.sortTemplate", async () => {
+            const sortType = await vscode.window.showQuickPick(getQuickPickItems(), { placeHolder: 'What do you want to sort?' });
+            if (sortType) {
+                await this.sortTemplate(sortType.value);
+            }
+        });
```

Comment:
```suggestion
        registerCommand("azurerm-vscode-tools.sortTemplate", async (_context: IActionContext, uri?: vscode.Uri) => {
            const sortType = await vscode.window.showQuickPick(getQuickPickItems(), { placeHolder: 'What do you want to sort?' });
            if (sortType) {
                await this.sortTemplate(sortType.value, uri);
            }
        });

```

Reply:
Best to use the Uri that's passed in whenever possible (e.g., right-click on editor -> Sort Template... will pass in the URI for the current document).  With the async nature of things, it's possible for the active editor to get changed by the time you retrieve it later.

Reply:
And even if it's not passed in, best to retrieve the active editor immediately rather than later.